pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go

```diff
@@ -817,58 +924,73 @@ func addPolicy(t *testing.T, cfg *memory.Controller, name, ns string, selector m
 		modify(&p)
 	}
 
-	_, err := cfg.Create(p)
+	_, err := s.cfg.Create(p)
 	if err != nil && strings.Contains(err.Error(), "item already exists") {
-		_, err = cfg.Update(p)
+		_, err = s.cfg.Update(p)
 	}
 	if err != nil {
 		t.Fatal(err)
 	}
 }
 
-func assertAddresses(t *testing.T, controller *FakeController, lookup string, names ...string) {
+func (s *ambientTestServer) deletePod(t *testing.T, name string) {
 	t.Helper()
-	want := sets.New(names...)
-	assert.EventuallyEqual(t, func() sets.String {
-		var addresses []*model.AddressInfo
-		if lookup == "" {
-			addresses = controller.ambientIndex.All()
-		} else {
-			addresses = controller.ambientIndex.Lookup(lookup)
-		}
-		have := sets.New[string]()
-		for _, address := range addresses {
-			switch addr := address.Address.Type.(type) {
-			case *workloadapi.Address_Workload:
-				have.Insert(addr.Workload.Name)
-			case *workloadapi.Address_Service:
-				have.Insert(addr.Service.Name)
-			}
-		}
-		return have
-	}, want, retry.Timeout(time.Second*3))
+	s.pc.Delete(name, testNS)
 }
 
-func deletePod(t *testing.T, pc clienttest.TestClient[*corev1.Pod], name string) {
-	t.Helper()
-	pc.Delete(name, "ns1")
-}
-
-func assertEvent(t *testing.T, fx *xdsfake.Updater, ip ...string) {
+func (s *ambientTestServer) assertEvent(t *testing.T, ip ...string) {
 	t.Helper()
 	want := strings.Join(ip, ",")
-	fx.MatchOrFail(t, xdsfake.Event{Type: "xds", ID: want})
+	s.fx.MatchOrFail(t, xdsfake.Event{Type: "xds", ID: want})
 }
 
-func deleteService(t *testing.T, sc clienttest.TestClient[*corev1.Service], name string) {
+func (s *ambientTestServer) deleteService(t *testing.T, name string) {
 	t.Helper()
-	sc.Delete(name, "ns1")
+	s.sc.Delete(name, testNS)
 }
 
-func addService(t *testing.T, sc clienttest.TestClient[*corev1.Service], name string, labels, annotations map[string]string,
+func (s *ambientTestServer) addService(t *testing.T, name string, labels, annotations map[string]string,
 	ports []int32, selector map[string]string, ip string,
 ) {
 	t.Helper()
-	service := generateService(name, "ns1", labels, annotations, ports, selector, ip)
-	sc.CreateOrUpdate(service)
+	service := generateService(name, testNS, labels, annotations, ports, selector, ip)
+	s.sc.CreateOrUpdate(service)
+}
+
+func (s *ambientTestServer) lookup(key string) []*model.AddressInfo {
+	if key == "" {
+		return s.controller.ambientIndex.All()
+	}
+	return s.controller.ambientIndex.Lookup(key)
+}
+
+func (s *ambientTestServer) clearEvents() {
+	s.fx.Clear()
+}
+
+// Returns the resource name for the given pod.
+func (s *ambientTestServer) pod(name string) string {
+	return fmt.Sprintf("%s//Pod/%s/%s",
+		s.controller.clusterID, testNS, name)
+}
```

Comment:
nit; i don't think this is quite flexible enough alone. I think we will want unit tests that write pods/WEs/SEs to other namespaces. for example, in https://github.com/istio/istio/pull/45621 i added some unit tests to ensure pods / WEs are only selected if they're in the same namespace as the selecting SE

perhaps we can just add another function here where we can provide the namespace as an override?

Reply:
also can we make the name here more specific? e.g. `podXdsName`?

Reply:
@kdorosh

> i don't think this is quite flexible enough alone.

I agree, but that's a general comment on all of these tests today. I think this should just be addressed holistically once someone needs to add a test for another namespace/network/cluster.